pyro/infer/elbo.py

```diff
@@ -25,9 +29,9 @@ class ELBO(object):
 
     def __init__(self,
                  num_particles=1,
-                 enum_discrete=False):
+                 max_iarange_nesting=float('inf')):
```

Comment:
is it weird to mix floats and ints?

Reply:
It seems pretty safe to me. We're taking care to consume this number only through comparisons (in functions like `sum_rightmost()` and `check_sites()` in #806 ) and we only read its value if it is less than some other finite number. This `float('inf')` solution seems cleaner to me than the alternatives like `INT_MAX` or `-1` for which we would need logic that is even less readable.